[feat] 지역(동네) 삭제 기능

pie2457 opened the pull request:
## Issues
- #47 

## What is this PR? 👓
지역(동네) 삭제 기능에 대한 PR입니다.

## Key changes 🔑
- 지역(동네) 삭제 기능
- 지역(동네) 삭제 기능 테스트
- 지역(동네) 추가시 기존에 선택한 동네와 새로 추가한 동네 모두 `isSelected = true`가 되서 **새로 추가한 동네를 true**로 **기존 동네를 false**로 변경하게끔 수정

pie2457 committed:
refactor #47 : 코드 리팩토링

- 잘못된 상수 이름 변경 MAXIMUM -> MINIMUM
- changeIsSelectedFalse() : 중복되는 로직 메서드로 추출
- getMember() : 메서드 삭제 -> 사용되지 않는 메서드를 로직으로 변경
- validateMemberTownSizeAndDuplicateRegion() : 새로운 지역(동네) 추가시 기존에 선택한 동네와 새로 추가한 동네 모두 isSelected = true가 되서 추가한 동네를 true로 기존 동네를 false로 변경하게끔 수정

pie2457 committed:
feat #47 : 에러코드 생성 및 핸들러 추가

pie2457 committed:
feat #47 : 지역(동네) 삭제 기능

pie2457 committed:
test #47 : 지역(동네) 삭제 기능 테스트

23Yong commented on src/main/java/pie/tomato/tomatomarket/application/memberTown/MemberTownService.java:
```diff
@@ -38,7 +39,8 @@ public class MemberTownService {
 	@Transactional
 	public void addMemberTown(Principal principal, AddMemberTownRequest memberTownRequest) {
-		Member member = getMember(principal);
+		Member member = memberRepository.findById(principal.getMemberId())
```
여기 `member` 변수 쓰이지 않는 것 같아요~

23Yong commented on src/main/java/pie/tomato/tomatomarket/application/memberTown/MemberTownService.java:
```diff
@@ -38,7 +39,8 @@ public class MemberTownService {
-		Member member = getMember(principal);
+		Member member = memberRepository.findById(principal.getMemberId())
+			.orElseThrow(() -> new NotFoundException(ErrorCode.NOT_FOUND_MEMBER));
 		Region region = regionRepository.findById(memberTownRequest.getAddressId())
```
여기 `region` 변수도 쓰이지 않는 것 같은데요~

23Yong commented on src/main/java/pie/tomato/tomatomarket/application/memberTown/MemberTownService.java:
```diff
@@ -58,18 +60,21 @@ private void validateMemberTownSizeAndDuplicateRegion(
 		}
 
-		if (memberTowns.size() > MEMBER_TOWN_MAXIMUM_SIZE) {
+		if (memberTowns.size() > MEMBER_TOWN_MINIMUM_SIZE) {
+			changeIsSelectedFalse(memberTowns, memberTownRequest.getAddressId());
 			throw new BadRequestException(ErrorCode.MAXIMUM_MEMBER_TOWN_SIZE);
```
최소 사이즈보다 큰데 false로 바꾸고 예외를 던져버리는게 이상한데요?

pie2457 replied:
제가 아주 멍청한짓을 했네요ㅎ,, 감사합니다 ! 바로 수정할게요 !

23Yong commented on src/main/java/pie/tomato/tomatomarket/presentation/memberTown/MemberTownController.java:
```diff
@@ -45,4 +47,11 @@ public ResponseEntity<CustomSlice<MemberTownListResponse>> findAll(@RequestParam
+	public ResponseEntity<Void> deleteMemberTown(@AuthPrincipal Principal principal,
+		@RequestBody DeleteMemberTownRequest deleteMemberTownRequest) {
+		memberTownService.deleteMemberTown(principal, deleteMemberTownRequest);
+		return ResponseEntity.ok().build();
```
200 OK 를 응답해도 되지만 204 도 한 번 고려해봐주세용

pie2457 replied:
넵 204가 더 알맞은거같아요 ! 감사합니당 👍🏼

23Yong reviewed (approved):
수고하셨습니다~

pie2457 committed:
refactor #47 : 리뷰사항 반영

- MemberTownService : membertown의 size가 1일시 기존 membertown의 isSelected를 false로 변경
- MemberTownController : 요청이 정상적으로 이루어졌을 시 응답코드를 204(noContent)로 변경
